### ISSUE TITLE
Disconnecting from the server may be delayed

### ISSUE THREAD

**Issue by waitwhatActo** [HomeworkSMP/server#1]
# Status: Resolved
Description: When player disconnects from server, it may take up to 1 minute for server to receive request. This issue may be intensified if player disconnected during combat.
Known Affected Versions: Java 1.21.x, Java 1.20.x

**Comment by waitwhatActo** [comment]
A fix will be attempted between Jan 1st 00:00 UTC - Jan 5th 00:00 UTC. 
During this time, the server will operate as usual.

**Comment by waitwhatActo** [comment]
# Issue Update
This issue is now under monitoring. ***Feel free to reopen if you encounter the same issue***.
## TL;DR
It is highly recommended to use a client that is the same version as the server. The issue may lie in the plugin ViaVersion. I have updated the server & plugins, upgraded specs and networking hardware in hopes of resolving this issue.

## *TL;* but you *D*o want to *R*ead
### Fixes
The test server used the same plugins as HomeworkSMP. It seems that ViaVersion may be causing this issue. I have updated the server from 1.21.3 to 1.21.4, and also related plugins. 
The server itself has been upgrade from i3-3320 to i7-1165G7, from 8GB DDR3 to 16GB DDR4. This change was accompanied by a home network upgrade, with a faster router (Ubiquiti UGC-Ultra), and a faster network switch.
### Tests
This issue has been tested "rigorously" on a test server.
The test server was on paper 1.21.4, and the client was using fabric 1.21.4, and the server displayed no issue disconnecting the player.
When the client dropped to 1.21.x, disconnecting from the server may sometimes be delayed slightly (avg 3s, max 20s).
When the client dropped to 1.16.x, the server disconnection is the same as 1.21.x.
When the client dropped to 1.8.9, the server displayed no issue disconnecting the player.

**Comment by waitwhatActo** [comment]
This issue is now resolved.
